developing_collections_structure.rs: Fix link formatting (#570)
ansible/ansible-documentation: docs/docsite/rst/dev_guide/developing_collections_structure.rst

Use the ``docs`` folder to describe how to use the roles and plugins the collection provides, role requirements, and so on.

> For certified collections, Automation Hub displays documents written in markdown in the main ``docs`` directory with no subdirectories. This will not display on `docs.ansible.com <https://docs.ansible.com>`_.

Reviewer: ```suggestion
For certified collections, Automation Hub displays documents written in markdown in the main ``docs`` directory with no subdirectories.
This documentation is not published to `docs.ansible.com <https://docs.ansible.com>`_.
```

Reviewer: Another good case for one sentence per line. No changes at all in the first sentence so it should not be part of the diff.

Reviewer: Also as a tip, always be sure to use "this + object" in documentation to refer to a specific thing.

Author: Thanks, i usually use it I think, it wasn't mine:)

Author: committed, PTAL
